### PR TITLE
make embed work for fq_default

### PR DESCRIPTION
**Pull request by tthsqe12**

### CODE REVIEW

**Comment by thofma** [comment]
Do we have some overlap?

**Comment by tthsqe12** [comment]
There are a few ways to find out.

File: src/flint/fq_default_embed.jl [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+function embed_matrices_pre(a::fq_default, b::fq_default, P::gfp_fmpz_poly)
+    k = parent(a)
+    K = parent(b)
+    m, n = degree(k), degree(K)
+    R = base_ring(P)
+    S1 = MatrixSpace(R, n, m)
+    S2 = MatrixSpace(R, m, n)
+    s1 = S1()
+    s2 = S2()
```
Comment (thofma) [inline src/flint/fq_default_embed.jl:128]:
```suggestion
    s1 = zero_matrix(R, n, m)
    s2 = zero_matrix(R, m, n)
```

File: src/flint/fq_default_embed.jl [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    _fq_default_embed_matrices(s1, s2, a, k, b, K, P)
+    return s1, s2
+end
+
+function embed_pre_mat(x::fq_default, K::FqDefaultFiniteField, M::gfp_fmpz_mat)
+
+    d = degree(parent(x))
+    S = MatrixSpace(base_ring(M), d, 1)
```
Comment (thofma) [inline src/flint/fq_default_embed.jl:137]:
```suggestion
    S = zero_matrix(base_ring(M), d, 1)
```

File: src/flint/fq_default_embed.jl [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            return s1, s2
+        end
+    end
+
+    a, b, P = embed_gens(k, K)
+    R = base_ring(P)
+    S1 = MatrixSpace(R, n, m)
+    S2 = MatrixSpace(R, m, n)
+    s1 = S1()
+    s2 = S2()
```
Comment (thofma) [inline src/flint/fq_default_embed.jl:114]:
```suggestion
    s1 = zero_matrix(R, n, m)
    s2 = zero_matrix(R, m, n)
```

**Pull request merged**